src/AspNetCoreModuleV2/AspNetCore/src/applicationinfo.cpp

```diff
@@ -267,46 +263,67 @@ APPLICATION_INFO::FindRequestHandlerAssembly()
 
         if (m_pConfiguration->QueryHostingModel() == APP_HOSTING_MODEL::HOSTING_IN_PROCESS)
         {
-            if (FAILED(hr = FindNativeAssemblyFromHostfxr(&struFileName)))
+            handlerDllName = g_pwzAspnetcoreInProcessRequestHandlerName;
+        }
+        else
+        {
+            handlerDllName = g_pwzAspnetcoreOutOfProcessRequestHandlerName;
+        }
+
+        // Try to see if RH is already loaded
+        g_hAspnetCoreRH = LoadLibraryW(handlerDllName);
```

Comment:
this will load the module. If somehow there is a assembly with the same name in some load path, this call will accidently loaded it. Please use GetModuleHandle https://msdn.microsoft.com/en-us/library/windows/desktop/ms683199(v=vs.85).aspx if you just want to make sure if the assembly has been loaded. Alos better perf